lib/puppet_x/puppetlabs/strings/yard/core_ext/yard.rb

```diff
@@ -0,0 +1,40 @@
+require 'yard'
+
+require 'puppet_x/puppetlabs/strings'
+
+# Patch the regular expression used to match namespaces
+# so it will allow namespace segments that begin with
+# both uppercase and lowercase letters (i.e. both
+# Puppet::Namespace and puppet::namespace)
+YARD::CodeObjects.send(:remove_const, :CONSTANTMATCH)
+YARD::CodeObjects::CONSTANTMATCH = /[a-zA-Z]\w*/
```

Comment:
I think these two lines should not be here. It means always setting the constant. A couple of lines down, you will then remove it, and set it again and the patch does not come into effect. (Or so it seems).